drivers/serial/uart_pl011.c

```diff
@@ -354,7 +354,8 @@ static int pl011_init(struct device *dev)
 {
 	int ret;
 	u32_t lcrh;
-	const struct uart_device_config *config = dev->config->config_info;
+	const struct uart_device_config *config __unused =
```

Comment:
why don't you just move this several lines down inside the #ifdef CONFIG_UART_INTERRUPT_DRIVEN instead of using __unused

Reply:
Because declaring the variable in the middle of the code inside the ifdef is horrible to me. Are you sure you want to do that?

Reply:
nvm, let's just use `DEV_CFG`.